### PR TITLE
WIP: Updates to the dch-photon docs

### LINKED ISSUE #1212: Explain why VIC doesn't support build or push
Per  some very good info from @pdaigle in https://vmware-com.socialcast.com/messages/37578087 (VMware internal), reproduced here:

The answer is two-fold. From an engineering standpoint build and push pose non-trivial challenges. Secondly, VIC initially focused on running existing images in production. As such, we focused on the opinionated, strong isolation provisioning model of the Container VM and worked on the assumption that image creation was happening somewhere else in the CI process. Throughout 2017 and the work we did with customer POCs, customers asked us to provide the ability to build and push and we came up with DCH. By brining the ephemeral quality of the containers to the container host, DCH allows us to a) eliminate snowflake deployments of the docker engine and b) promote efficient resource consumption by providing an easy mechanism for provision/tear down of a container engine that fits well with CI automation.

The workflow for the app re-packaging use case would be:

- Instantiate DCH to build/push the image to the private registry;
- Remove DCH;
- Pull/run image from VCH in production;

Because of the ephemeral quality of the DCH and because it is itself a container image, this can easily be scripted or integrated with an existing CI tool (for example, customers could leverage Jenkins for this).
Another common use case for application re-packaging is Commercial Off-The-Shelf (COTS) software. We see more and more software publishers leveraging the docker image as a distribution mechanism and VIC fits well in that model.

### LINKED ISSUE #1759: Corrections required in dch-photon procedure
@stuclem commented on [Wed May 16 2018](https://github.com/vmware/vic/issues/7955)

Per internal Bugzilla bug:

Documentation does not clearly explain how to deploy dch-photon. 
This is 3 step process and should be presented as such
1. Deploy DCH-VCH (https://vmware.github.io/vic-product/assets/files/html/1.3/vic_vsphere_admin/deploy_vch_dchphoton.html). Under vic-machine command public network is not specified in the example and it is needed.
2. Create container and map ports 12375:2375 for non TLS. (This part is nowhere in the documentation, only TLS with mapping 12376:2376 documentation is there - https://vmware.github.io/vic-product/assets/files/html/1.3/vic_app_dev/dchphoton_options.html)
3. Start container and start using docker -H <VCH_DCH>:12375 ( https://vmware.github.io/vic-product/assets/files/html/1.3/vic_app_dev/test_photon.html) this refers to step 2, whcih is not defined anywhere in the documentation.

---

@stuclem commented on [Wed May 16 2018](https://github.com/vmware/vic/issues/7955#issuecomment-389516813)

Oops, opened this in the wrong repo!

bug2116679

### LINKED ISSUE #1761: Document supported version of dch-photon
From internal bugzilla bug: 

Documentation does not specify which versions of dch-photon are supported. 
This page - https://vmware.github.io/vic-product/assets/files/html/1.3/vic_app_dev/configure_docker_client.html shows that Docker client (which corresponds to dch-photon version) ver 1.13 is supported. I do not think that is enough and definitely not clear.

### LINKED ISSUE #2049: Enforce nomenclature and capitalization standards in the docs
We need to check how well we apply standards of product naming and capitalization. For example:

- vSphere Integrated Containers Engine. This is a product name, therefore capitalized. Subsequently, “engine” or “the engine”. Lower-case, as this is not a product name.
- “Docker container host” (lower case), but I just looked at the Docker docs and they just use “Docker host”: https://docs.docker.com/network/host/.
- “Docker Engine” is capitalized in the Docker docs: https://docs.docker.com/engine/
- By contrast, the Docker docs do not capitalize “Docker client”: https://docs.docker.com/engine/reference/commandline/cli/

### PR DESCRIPTION
**Pull request by stuclem**
Fixes #1761, fixes #1759, fixes #1212, fixes #2049.

### CODE REVIEW

**Comment by stuclem** [comment]
@renmaosheng would it be possible for someone from development or QE to work through the procedures under https://vmware.github.io/vic-product/assets/files/html/1.4/vic_app_dev/build_push_images.html to verify that they do indeed work? I don't think that this procedure was ever tested by anyone other than myself.

**Comment by renmaosheng** [comment]
@yuyangbj could you please take a look? thanks.

**Comment by stuclem** [comment]
@yuyangbj thank you in advance. To make things easier, here are the links to the latest versions of the docs:

- [Deploy a Virtual Container Host with a Volume Store and vSphere Integrated Containers Registry Access](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_vsphere_admin/deploy_vch_dchphoton.md)
- [Building and Pushing Images with the dch-photon Docker Engine](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/build_push_images.md)
- [Add the Registry Certificate to a Custom Image](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/photon_cert_custom.md)
- [Manually Add the Registry Certificate to a dch-photon Container VM](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/photon_cert_manual.md)
- [Build, Push, and Pull an Image with dch-photon](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/test_photon.md)
- [Advanced dch-photon Deployment](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/dchphoton_options.md)

Thank you again!

**Comment by stuclem** [comment]
https://github.com/vmware/vic-product/pull/1989/commits/634dfcf65e2b6bc7a5111f6e7d264c972e6a2f26 fixes https://github.com/vmware/vic-product/issues/1212

**Comment by stuclem** [comment]
https://github.com/vmware/vic-product/pull/1989/commits/5f6e3e9fb8be81748e29babae368ac2be6d232ab fixes https://github.com/vmware/vic-product/issues/2049.

**Comment by stuclem** [comment]
https://github.com/vmware/vic-product/pull/1989/commits/0cab1e84be20c9cfd33e2feb941ca02893d1fb3e is towards https://github.com/vmware/vic-product/issues/1760.

**Comment by stuclem** [comment]
https://github.com/vmware/vic-product/pull/1989/commits/d6b80ca40ace59a5647289d9b6ef9d6e1b2b6734 fixes https://github.com/vmware/vic-product/issues/1932. Thanks to @farflungfish for this contribution, and apologies for the delay in getting it into the docs.

**Comment by stuclem** [comment]
This PR also fixes https://github.com/vmware/vic-product/issues/1760

**Comment by stuclem** [comment]
This PR also fixes https://github.com/vmware/vic-product/issues/948.

**Comment by stuclem** [comment]
Fixes #2113 and fixes #969

**Comment by stuclem** [comment]
@yuyangbj I have tested all of the `dch-photon` procedures and have made them work. I have also rewritten the docs to hopefully make things clearer.

To make things easier, here are the links to the latest versions of the docs:

- [Building and Pushing Images with the dch-photon Docker Engine](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/build_push_images.md)
  - [Deploy a Test dch-photon Instance](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/deploy_test_dch.md)
     - [Add the Registry Certificate to a Custom Image](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/photon_cert_custom.md)
     - [Manually Add the Registry Certificate to a dch-photon Container VM](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/photon_cert_manual.md)
     - [Build, Push, and Pull an Image with dch-photon](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/test_photon.md)
  - [Advanced Use of dch-photon](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/dch_advanced.md)
     - [Advanced dch-photon Deployment](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/dchphoton_options.md)
     - [Expand the Root Disk on a dch-photon Docker Engine](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/dch_expand_disk.md)
     - [Configure dch-photon to Use Proxy Servers](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/dch_photon_proxy.md)
     - [Add a Custom Registry Certificate Authority to dch-photon](https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/add_custom_ca.md)

You need to work through the documents in order, as one procedure follows the next. Thank you in advance!

File: docs/user_doc/vic_app_dev/build_push_images.md [hunk 1]
```diff
@@ -33,15 +49,12 @@ The purpose of `dch-photon` is primarily to build images and push them to regist
 
 You can provide the certificate to `dch-photon` in one of two ways:
 
--  Build a custom image that has the certificate embedded in it, as described in [Add the Registry Certificate to a Custom Image](photon_cert_custom.md). This method is preferable since you only need to perform the operation once.
--  Manually copy the certificate in to a `dch-photon` container running in a VCH by using `docker cp`, as described in [Manually Add the Registry Certificate to a `dch-photon` VM](photon_cert_manual.md).
-
-When you have deployed `dch-photon` with the registry certificate, you can use it to build an image and push that image from `dch-photon` to vSphere Integrated Containers Registry. You can then pull the image from the registry into a VCH for deployment. For information about building, pushing, and pulling an image, see [Build, Push, and Pull an Image with `dch-photon`](test_photon.md).
+-  Build a custom `dch-photon` image that has the certificate embedded in it. This method is preferable since you only need to perform the operation once.
```
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_app_dev/build_push_images.md]: @yuyangbj  this is just the introduction, which lists the main steps involved in using dch-photon. The examples of how to actually do this are in https://github.com/stuclem/vic-product/blob/dchphoton-update/docs/user_doc/vic_app_dev/deploy_test_dch.md

**Comment by stuclem** [comment]
@vidya-v can you please review/approve this one so that I can merge it? 

@yuyangbj we can continue the review outside of this PR. 

Thanks!

**Comment by stuclem** [comment]
Thanks @vidya-v!

**Pull request merged**